src/ssl.c

```diff
@@ -14113,51 +14168,59 @@ WOLFSSL_SESSION* GetSessionClient(WOLFSSL* ssl, const byte* id, int len)
         return NULL;
     }
 
-    if (wc_LockMutex(&session_mutex) != 0) {
-        WOLFSSL_MSG("Lock session mutex failed");
+    /* start from most recently used */
+    if (wc_LockMutex(&clisession_mutex) != 0) {
         return NULL;
     }
-
-    /* start from most recently used */
     count = min((word32)ClientCache[row].totalCount, SESSIONS_PER_ROW);
     idx = ClientCache[row].nextIdx - 1;
-    if (idx < 0)
+    if (idx < 0 || idx >= SESSIONS_PER_ROW) {
         idx = SESSIONS_PER_ROW - 1; /* if back to front, the previous was end */
+    }
+    clSess = ClientCache[row].Clients;
 
     for (; count > 0; --count) {
         WOLFSSL_SESSION* current;
-        ClientSession   clSess;
+        SessionRow* sessRow;
 
-        if (idx >= SESSIONS_PER_ROW || idx < 0) { /* sanity check */
-            WOLFSSL_MSG("Bad idx");
-            break;
+        if (clSess[idx].serverRow >= SESSION_ROWS) {
+            WOLFSSL_MSG("Client cache serverRow invalid");
+            return NULL;
```

Comment:
Need to unlock clisession_mutex.
break like below?

Reply:
Fixed